rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/service/LibraryService.kt

```diff
@@ -1,35 +1,54 @@
 package systems.ajax.englishstudytelegrambot.service
 
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import systems.ajax.englishstudytelegrambot.dto.entity.LibraryDtoResponse
+import systems.ajax.englishstudytelegrambot.dto.entity.WordDtoResponse
+import systems.ajax.englishstudytelegrambot.dto.entity.toDtoResponse
 import systems.ajax.englishstudytelegrambot.entity.Library
 import systems.ajax.englishstudytelegrambot.entity.Word
+import systems.ajax.englishstudytelegrambot.exception.LibraryIsMissingException
+import systems.ajax.englishstudytelegrambot.exception.LibraryWithTheSameNameForUserAlreadyExistException
 import systems.ajax.englishstudytelegrambot.repository.LibraryRepository
 
 interface LibraryService {
 
-    fun createNewLibrary(nameOfNewLibrary: String, telegramUserId: String): Library
+    fun createNewLibrary(nameOfNewLibrary: String, telegramUserId: String): Mono<LibraryDtoResponse>
 
-    fun deleteLibrary(nameOfLibraryForDeleting: String, telegramUserId: String): Library
+    fun deleteLibrary(nameOfLibraryForDeleting: String, telegramUserId: String): Mono<LibraryDtoResponse>
 
-    fun getAllWordsFromLibrary(libraryName: String, telegramUserId: String): List<Word>
+    fun getAllWordsFromLibrary(libraryName: String, telegramUserId: String): Flux<WordDtoResponse>
 }
 
 @Service
 class LibraryServiceImpl(
     val libraryRepository: LibraryRepository
 ) : LibraryService {
 
-    override fun createNewLibrary(nameOfNewLibrary: String, telegramUserId: String): Library =
-        libraryRepository.saveNewLibrary(nameOfNewLibrary, telegramUserId)
-
-    override fun deleteLibrary(nameOfLibraryForDeleting: String, telegramUserId: String): Library {
-        val libraryId =
-            libraryRepository.getLibraryIdByLibraryNameAndTelegramUserId(nameOfLibraryForDeleting, telegramUserId)
-        return libraryRepository.deleteLibrary(libraryId)
-    }
-
-    override fun getAllWordsFromLibrary(libraryName: String, telegramUserId: String): List<Word> {
-        val libraryId = libraryRepository.getLibraryIdByLibraryNameAndTelegramUserId(libraryName, telegramUserId)
-        return libraryRepository.getAllWordsFromLibrary(libraryId)
-    }
+    override fun createNewLibrary(nameOfNewLibrary: String, telegramUserId: String): Mono<LibraryDtoResponse> =
+        libraryRepository
+            .saveNewLibrary(nameOfNewLibrary, telegramUserId)
+            .onErrorMap{LibraryWithTheSameNameForUserAlreadyExistException()}
```

Comment:
```suggestion
    override fun createNewLibrary(nameOfNewLibrary: String, telegramUserId: String): Mono<LibraryDtoResponse> =
        libraryRepository
            .saveNewLibrary(nameOfNewLibrary, telegramUserId)
            .onErrorMap{ LibraryWithTheSameNameForUserAlreadyExistException() }
```
If you don't pass any argument to exception, consider making it object, so you won't create the same instance each time
Alternatively, pass original error as source